sharktank/sharktank/types/tensors.py

```diff
@@ -752,10 +752,29 @@ class ShardedTensor(InferenceTensor):
     """
 
     def __init__(
-        self, *, shape: list[int], shard_dim: int | None, name: str = UnnamedTensorName
+        self, *, 
+        shape: list[int], 
+        shard_dim: int | None, 
+        name: str = UnnamedTensorName,
+        devices: Tuple[int],
+        devices_pinned: bool,
     ):
         super().__init__(name=name, shape=shape)
         self.shard_dim = shard_dim
+        self._devices = devices
+        self._devices_pinned = devices_pinned
```

Comment:
We can probably just rename this to `pinned`. Its more succinct and conceptually its basically the only definition is memory pinning.